src/internal/operators/delay.ts

```diff
@@ -74,88 +73,41 @@ class DelayOperator<T> implements Operator<T, T> {
   }
 }
 
-interface DelayState<T> {
-  source: DelaySubscriber<T>;
-  destination: PartialObserver<T>;
-  scheduler: SchedulerLike;
-}
-
 /**
  * We need this JSDoc comment for affecting ESDoc.
  * @ignore
  * @extends {Ignored}
  */
 class DelaySubscriber<T> extends Subscriber<T> {
-  private queue: Array<DelayMessage<T>> = [];
-  private active: boolean = false;
-  private errored: boolean = false;
-
-  private static dispatch<T>(this: SchedulerAction<DelayState<T>>, state: DelayState<T>): void {
-    const source = state.source;
-    const queue = source.queue;
-    const scheduler = state.scheduler;
-    const destination = state.destination;
-
-    while (queue.length > 0 && (queue[0].time - scheduler.now()) <= 0) {
-      queue.shift().notification.observe(destination);
-    }
-
-    if (queue.length > 0) {
-      const delay = Math.max(0, queue[0].time - scheduler.now());
-      this.schedule(state, delay);
-    } else {
-      this.unsubscribe();
-      source.active = false;
-    }
-  }
-
   constructor(destination: Subscriber<T>,
               private delay: number,
               private scheduler: SchedulerLike) {
     super(destination);
   }
 
-  private _schedule(scheduler: SchedulerLike): void {
-    this.active = true;
-    const destination = this.destination as Subscription;
-    destination.add(scheduler.schedule<DelayState<T>>(DelaySubscriber.dispatch, this.delay, {
-      source: this, destination: this.destination, scheduler: scheduler
-    }));
-  }
-
   private scheduleNotification(notification: Notification<T>): void {
-    if (this.errored === true) {
-      return;
-    }
-
-    const scheduler = this.scheduler;
-    const message = new DelayMessage(scheduler.now() + this.delay, notification);
-    this.queue.push(message);
-
-    if (this.active === false) {
-      this._schedule(scheduler);
-    }
+    const destination = this.destination as Subscriber<T>;
+    this.scheduler.schedule<DelayState<T>>(dispatch, this.delay, { notification, destination });
```

Comment:
What happens with the subscription that's returned here? What if the subscriber to the observable returned by `delay` unsubscribes? Shouldn't any pending notifications be unsubscribed in that situation?